repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractVersionUtils.java

```diff
@@ -0,0 +1,85 @@
+import org.apache.atlas.model.discovery.IndexSearchParams;
+import org.apache.atlas.model.instance.AtlasEntity;
+import org.apache.atlas.model.instance.AtlasEntityHeader;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.store.graph.AtlasEntityStore;
+import org.apache.atlas.repository.store.graph.v2.*;
+import org.apache.atlas.type.AtlasTypeRegistry;
+
+import java.util.*;
+
+import static org.apache.atlas.repository.Constants.*;
+import static org.apache.atlas.repository.util.AtlasEntityUtils.mapOf;
+
+
+public class ContractVersionUtils {
+
+    private final EntityMutationContext context;
+    public final EntityGraphRetriever entityRetriever;
+    private final AtlasTypeRegistry atlasTypeRegistry;
+    private AtlasEntityStore entityStore;
+    private String contractQName;
+
+    public final AtlasGraph graph;
+    private List<AtlasEntityHeader> versionList;
+    private EntityDiscoveryService discovery;
+
+    public ContractVersionUtils(String contractQName, EntityMutationContext context, EntityGraphRetriever entityRetriever,
+                                AtlasTypeRegistry atlasTypeRegistry, AtlasEntityStore entityStore, AtlasGraph graph,
+                                EntityDiscoveryService discovery) {
+        this.context = context;
+        this.entityRetriever = entityRetriever;
+        this.atlasTypeRegistry = atlasTypeRegistry;
+        this.graph = graph;
+        this.entityStore = entityStore;
+        this.contractQName = contractQName;
+        this.discovery = discovery;
+    }
+
+    private void extractAllVersions() throws AtlasBaseException {
+        List<AtlasEntityHeader> ret = new ArrayList<>();
+
+        IndexSearchParams indexSearchParams = new IndexSearchParams();
+        Map<String, Object> dsl = new HashMap<>();
+
+        List mustClauseList = new ArrayList();
+        mustClauseList.add(mapOf("term", mapOf("__typeName.keyword", CONTRACT_ENTITY_TYPE)));
+        mustClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, String.format("%s/*", contractQName))));
+
+        dsl.put("query", mapOf("bool", mapOf("must", mustClauseList)));
+        Set<String> attributes = new HashSet<>();
+        attributes.add(ATTR_CONTRACT);
+        attributes.add(ATTR_CERTIFICATE_STATUS);
+
+        indexSearchParams.setDsl(dsl);
+        indexSearchParams.setAttributes(attributes);
+        indexSearchParams.setSuppressLogs(true);
+
+        AtlasSearchResult result = discovery.directIndexSearch(indexSearchParams);
+        if (result != null) {
+            ret = result.getEntities();
+        }
+        this.versionList = ret;
+    }
+
+    public AtlasEntity getCurrentVersion() throws AtlasBaseException {
+        if (this.versionList == null) {
+            extractAllVersions();
+        }
+        if (this.versionList == null) {
+            return null;
+        }
+        Collections.sort(this.versionList, (e1, e2) -> {
```

Comment:
instead of sorting in-memory, can we rely on Elasticsearch query sort?
inside extractAllVersions(), we can sort using `__timestamp` attribute which is the time of asset creation

Reply:
Also, this comparison will be based on a string so imagine you have V1, V2, V11
I fear sorting will generate ordering V2, V11, V1
we end up getting V2 as the latest version

Reply:
And consider limiting ES query result size to 1

Reply:
sorting using dataContractVersion attribute as this is long type attribute.